### PR TITLE
Fix S2 ComboBox and NumberField custom width

### PR DESCRIPTION
**Pull request by devongovett**

### CODE REVIEW

**Review by LFDanLu — APPROVED** [review]
LGTM, verified the ComboBox and NumberField custom widths are being properly used and the other stories are sized appropriately

File: packages/@react-spectrum/s2/stories/ComboBox.stories.tsx [hunk 1]
```diff
@@ -210,3 +211,11 @@ ContextualHelpExample.parameters = {
     }
   }
 };
+
+export const CustomWidth = (args: any) => <Example {...args} styles={style({width: 384})} />;
```
Comment (snowystinger) [inline packages/@react-spectrum/s2/stories/ComboBox.stories.tsx:215]: just a heads up for the future, I'm pretty sure the way Storybook wants you to extend these is to instead do something like 
```
export const CustomWidth = {
  ...Example
  args: {
    ...Example.args,
    styles: style({width: 384})
  }
}
```

**Review by snowystinger — APPROVED** [review]
looks good

File: packages/@react-spectrum/s2/stories/ComboBox.stories.tsx (hunk 1 above)
Comment (devongovett) [reply packages/@react-spectrum/s2/stories/ComboBox.stories.tsx:215]: jsx is so much more readable though. what's the advantage of this?

**Pull request merged**